AWSSignature4DynamicValue.js

```diff
@@ -253,7 +253,13 @@ var AWSSignature4DynamicValue = function() {
           })
         }
 
-        headers['host'] = uri.host.toLowerCase();
+        if (headers.host) {
+            // Use the host header if set
+            headers.host = headers.host.toLowerCase();
+        } else {
+            // Otherwise, extract it from the target uri
+            headers.host = uri.host.toLowerCase();
+        }
```

Comment:
Github doesn't let me comment down there, but the `target` variable in line #278 is never used.

Reply:
Thanks, that's just unused code from an earlier version. I'll remove it on the next commit.